server/runtime/src/main/java/org/infinispan/server/security/UserTool.java

```diff
@@ -0,0 +1,192 @@
+package org.infinispan.server.security;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.infinispan.Version;
+import org.infinispan.server.Server;
+import org.infinispan.server.tool.Main;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 10.0
+ **/
+public class UserTool extends Main {
+   public static final String DEFAULT_USERS_PROPERTIES_FILE = "users.properties";
+   public static final String DEFAULT_GROUPS_PROPERTIES_FILE = "groups.properties";
+   public static final String DEFAULT_REALM_NAME = "default";
+
+   private String username = null;
+   private String password = null;
+   private String realm = DEFAULT_REALM_NAME;
+   private String usersFileName = DEFAULT_USERS_PROPERTIES_FILE;
+   private String groupsFileName = DEFAULT_GROUPS_PROPERTIES_FILE;
+   private List<String> addGroups = new ArrayList<>();
+   private boolean batchMode = false;
+
+   public static void main(String args[]) {
+      UserTool userTool = new UserTool();
+      userTool.run(args);
+   }
+
+   protected void handleArgumentCommand(String command, String parameter, Iterator<String> args) {
+      switch (command) {
```

Comment:
We can also just call `if (parameter == null) parameter = args.next()` once here.

Reply:
No, because some commands take no parameters, e.g. -h, -v

Reply:
That's in Main.java, not the impl.